core/platform.js

```diff
@@ -75,23 +75,26 @@ Platform.prototype.messageRxd = function(api, event) {
 
 Platform.prototype.setModes = function(modes) {
     try {
+        var i = 0;
```

Comment:
You do realize the scope of this will be overridden by the var in the if statement right?